File: Java/7-ReverseInteger.java
public class Solution {
    public int reverse(int x) {
        int sign = x >= 0 ? 1 : -1;
        x = Math.abs(x);
        int rev = 0;
        while (x != 0){
            int tail = x % 10;
            int newRes = rev * 10 + tail;
            if ((newRes - tail) / 10 != rev){
                return 0; 
            }
            rev = newRes;
            x = x / 10;
        }
        return (sign * rev);
    }
}

File: cpp/7-ReverseInteger.cpp
//156323
//323651
class Solution {
public:
    int reverse(int x) {
        int res = 0;
        int sign = 1;
        if (x < 0) {
            sign = -1;
            x *= sign;
        }
        while (x > 0) {
            int tmp = res * 10 + x % 10;
            if (tmp / 10 != res) {
                return 0;
            }
            res = tmp;
            x /= 10;
        }
        return res * sign;
    }
};
